File: src/app/models/CPU.ts
import {Product} from './Product';
import {Socket} from './Socket';

export interface CPU extends Product {
  socket: Socket;
  model: string;
  frequencyGHz: number;
  cores: number;
  threads: number;
  integratedGraphicsModel: string | null;
  tdpWatts: number;
}

File: backend/src/main/java/bg/pcbudget/backend/models/CPU.java
package bg.pcbudget.backend.models;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "cpus")
@Getter
@Setter
@NoArgsConstructor
public class CPU extends Product {

  @ManyToOne
  @JoinColumn(name = "socket_id")
  private Socket socket;

  private String model;
  private Double frequencyGHz;
  private Integer cores;
  private Integer threads;
  private String integratedGraphicsModel;
  private Integer tdpWatts;
}
